lib/src/common/repos_service.dart

```diff
@@ -208,86 +247,192 @@ class RepositoriesService extends Service {
+      "GET",
+      "/repos/${slug.fullName}/commits/${commit.sha}/comments",
+      (i) => CommitComment.fromJSON(i),
+      statusCode: StatusCodes.OK,
+    );
   }
 
-  Future<bool> removeCollaborator(RepositorySlug slug, String user) {
-    return _github
-        .request("DELETE", "/repos/${slug.fullName}/collaborators/$user")
-        .then((response) {
-      return response.statusCode == 204;
+  /// Returns a list of all commit comments in a repository.
+  ///
+  /// https://developer.github.com/v3/repos/comments/#list-comments-for-a-single-commit
+  Stream<CommitComment> listCommitComments(RepositorySlug slug) {
+    return PaginationHelper(_github)
+        .objects<Map<String, dynamic>, CommitComment>(
+      "GET",
+      "repos/${slug.fullName}/comments",
+      (i) => CommitComment.fromJSON(i),
+      statusCode: StatusCodes.OK,
+    );
+  }
+
+  /// Create a comment for a commit using its sha.
+  /// * [body]: The contents of the comment.
+  /// * [path]: Relative path of the file to comment on.
+  /// * [position]: Line index in the diff to comment on.
+  /// * [line]: **Deprecated**. Use position parameter instead. Line number in the file to comment on.
+  ///
+  /// https://developer.github.com/v3/repos/comments/#create-a-commit-comment
+  Future<CommitComment> createCommitComment(
+    RepositorySlug slug,
+    RepositoryCommit commit, {
+    @required String body,
+    String path,
+    int position,
+    @Deprecated('Use position parameter instead') int line,
+  }) async {
+    final Map<String, dynamic> data = createNonNullMap({
+      'body': body,
+      'path': path,
+      'position': position,
+      'line': line,
     });
+    return _github.postJSON<Map<String, dynamic>, CommitComment>(
+      "/repos/${slug.fullName}/commits/${commit.sha}/comments",
+      body: jsonEncode(data),
+      statusCode: 201,
+      convert: (i) => CommitComment.fromJSON(i),
+    );
   }
 
-  // TODO: Implement listComments: https://developer.github.com/v3/repos/comments/#list-commit-comments-for-a-repository
-  // TODO: Implement listCommitComments: https://developer.github.com/v3/repos/comments/#list-comments-for-a-single-commit
-  // TODO: Implement createComment: https://developer.github.com/v3/repos/comments/#create-a-commit-comment
-  // TODO: Implement getComment: https://developer.github.com/v3/repos/comments/#get-a-single-commit-comment
-  // TODO: Implement updateComment: https://developer.github.com/v3/repos/comments/#update-a-commit-comment
-  // TODO: Implement deleteComment: https://developer.github.com/v3/repos/comments/#delete-a-commit-comment
+  /// Retrieve a commit comment by its id.
+  ///
+  /// https://developer.github.com/v3/repos/comments/#get-a-single-commit-comment
+  Future<CommitComment> getCommitComment(RepositorySlug slug,
+      {@required int id}) async {
+    return _github.getJSON<Map<String, dynamic>, CommitComment>(
+      "/repos/${slug.fullName}/comments/id",
```

Comment:
Should this insert the id into the string like this? (note the dollar sign `$`)
```suggestion
      "/repos/${slug.fullName}/comments/$id",
```

Reply:
You're right!